tools/src/bin/scip-indexer.rs

```diff
@@ -774,7 +855,12 @@ fn analyze_using_scip(
                 // so only process the first element.
                 if let Some(rel) = scip_sym_info.relationships.first() {
                     if let Some(rel_sinfo) = scip_symbol_to_structured.get(&rel.symbol) {
```

Comment:
This assumes that the SymbolInformation of the supers/overrides has already been seen, which isn't necessarily the case. For instance with that code no class relationship information is generated:
```java
class Derivative extends Base {}
class Base {}
```

This could be fixed in a number of ways:
1. Extracting the processing of relationships in another pass to make sure that `scip_symbol_to_structured` has been properly populated already.
2. Doing like in 57ca05a2 and adding entries to `scip_symbol_to_structured` as needed, and when processing the relevant `SymbolInformation` again updating the data if an empty entry was already created. `Structured{Super,Override}Info::pretty` may get out of sync.
3. Only keep the symbol in `Structured{Super,Override}Info`, that removes the issue of getting out of sync altogether.

My preferred option is option 3, because I don't see where the `Structured{Super,Override}Info::pretty` and `StructuredSuperInfo::props` are currently used anyway, and that's duplicated data that belongs more to the target than to the relationship (unless the objective is to use relative paths like `super::SomeTrait`).

Reply:
Option 3 is fine.  The current structured record type is largely just what my first draft of the structured representation was for C++ and was biased towards what clang provided and some JS-based prototyping I was doing for diagramming benefited from[1].  We definitely do not need to populate things at analysis/indexing time that can be handled by the cross-reference process.

The "pretty" is something I think we can move towards removing/eliminating since it does canonically come from the symbol definition itself and we do largely ignore it.  (And indeed, for the reciprocal relationships to super and override, we only emit the symbol.)

The "props" is something I would like to retain, but it only needs to be used for cases where the information is specific to the relationship between the current symbol and the upward-edge to the super/override.  There is no need to propagate information from the related symbol to the current symbol as part of indexing/analysis.  If we end up wanting that information on the current symbol for runtime performance reasons, we can do that during the crossref process as well.

1: In my JS prototype, diagramming happened client-side and with us having to perform a network request per symbol lookup, so there was a bias towards including information about the targets of edges in a symbol since it could potentially provide an order of magnitude of reduction in diagram lookups.  These are not concerns we have anymore.